### DISCUSSION TITLE
en_core_web_trf training for NER task

### DISCUSSION THREAD

**Discussion by sureshPalla230393** [explosion/spaCy#8213] [Help: Coding & Implementations] [answered]
Hi, 
(i) Are the below pipelines the ones which to be trained for NER task or did I miss anything else?
`pipe_exceptions = ["ner", "trf_wordpiecer", "trf_tok2vec"] `

(ii) Can anyone please elaborate -  "trf_wordpiecer" and "trf_tok2vec" and "transformer" components of the 'en_core_web_trf'?
I don't understand what does each component do for the NER task.

**Comment by polm** [comment, marked as answer]
I'm not sure but it sounds like you're maybe using spaCy v2? 

I would suggest you use the quickstart and follow the instructions there. It will guide you through training NER with Transformers. 

https://spacy.io/usage/training#quickstart

You may also find the example projects useful.

https://github.com/explosion/projects/tree/v3/tutorials

**Reply by sureshPalla230393** [reply]
@polm , I have updated to Spacy v3 and now i have a question on which loss function is used to calculate `transformer` component loss. To my understanding from the Spacy v3 documentation when training for NER task the NER component loss function is used to backpropagate the gradients. But when i train the `en_core_web_trf` for NER task, i see `transformer` loss and `NER` loss separately. Can you please explain what is this `transformer` loss.
![image](https://user-images.githubusercontent.com/33569360/120101066-d6859800-c161-11eb-991f-776b0a943c17.png)

**Reply by polm** [reply]
Please do not ask the same question in multiple places, it is not helpful. We will reply to you eventually. For reference this is where you first asked this question. https://github.com/explosion/spaCy/discussions/8211
